htdocs/fichinter/class/fichinterligne.class.php

```diff
@@ -0,0 +1,392 @@
+<?php
+/* Copyright (C) 2002-2003 Rodolphe Quiedeville <rodolphe@quiedeville.org>
+ * Copyright (C) 2004-2014 Laurent Destailleur  <eldy@users.sourceforge.net>
+ * Copyright (C) 2005-2012 Regis Houssin        <regis.houssin@inodbox.com>
+ * Copyright (C) 2011-2020 Juanjo Menent        <jmenent@2byte.es>
+ * Copyright (C) 2015      Marcos García        <marcosgdf@gmail.com>
+ * Copyright (C) 2015-2020 Charlene Benke       <charlie@patas-monkey.com>
+ * Copyright (C) 2018      Nicolas ZABOURI	    <info@inovea-conseil.com>
+ * Copyright (C) 2018-2024  Frédéric France     <frederic.france@free.fr>
+ * Copyright (C) 2023-2024  William Mead        <william.mead@manchenumerique.fr>
+ * Copyright (C) 2024		MDW					<mdeweerd@users.noreply.github.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * 	\file       htdocs/fichinter/class/fichinterligne.class.php
+ * 	\ingroup    fichinter
+ * 	\brief      File for class to manage intervention lines
+ */
+require_once DOL_DOCUMENT_ROOT.'/core/class/commonobjectline.class.php';
+
+/**
+ *	Class to manage intervention lines
+ */
+class FichinterLigne extends CommonObjectLine
+{
+	/**
+	 * @var DoliDB Database handler.
+	 */
+	public $db;
+
+	/**
+	 * @var string Error code (or message)
+	 */
+	public $error = '';
+
+	// From llx_fichinterdet
+	/**
+	 * @var int ID
+	 */
+	public $fk_fichinter;
+
+	public $desc; 		// Description ligne
+
+	/**
+	 * @var int Date of intervention
+	 */
+	public $date; 		// Date intervention
+	/**
+	 * @var int Date of intervention
+	 * @deprecated
+	 */
+	public $datei; 		// Date intervention
+
+	public $duration; 	// Duration of intervention
+	public $rang = 0;
+	public $tva_tx;
```

Comment:
These phan notifications appear because there are no exceptions for the new file.
To fix the PhanUnknownPropertyType, a phpdoc should be added.

Here (from CommonInvoice) that could be:
```
    /**
     * VAT %  Vat rate can be like "21.30 (CODE)"
     * @var string|float
     */
    public $tva_tx;
```

These also appear with PHPSTAN as "Property FichinterLigne::$tva_tx has no type specified.".

You can filter these notices on https://cti.dolibarr.org/ by looking for  `FichinterLigne`  or `fichinter.class`

Reply:
Thanks for help